components/service-operator/apis/database/v1beta1/postgres_types.go

```diff
@@ -16,60 +16,213 @@ limitations under the License.
+		Engine:                      Engine,
+		MasterUsername:              masterUsernameSecretRef,
+		MasterUserPassword:          masterPasswordSecretRef,
+		DBClusterParameterGroupName: cloudformation.Ref(PostgresResourceClusterParameterGroup),
+		Tags:                        tags,
+		VpcSecurityGroupIds: []string{
+			cloudformation.Ref(VPCSecurityGroupIDParameterName),
+		},
+		DBSubnetGroupName: cloudformation.Ref(DBSubnetGroupNameParameterName),
+	}
+
+	template.Resources[PostgresResourceMasterCredentialsAttachment] = &cloudformation.AWSSecretsManagerSecretTargetAttachment{
+		SecretId:   cloudformation.Ref(PostgresResourceMasterCredentials),
+		TargetId:   cloudformation.Ref(PostgresResourceCluster),
+		TargetType: "AWS::RDS::DBCluster",
+	}
+
+	instanceCount := p.Spec.AWS.InstanceCount
+	if instanceCount < 1 {
+		instanceCount = DefaultInstanceCount
+	}
+	for i := 0; i < instanceCount; i++ {
+		template.Resources[fmt.Sprintf("%s%d", PostgresResourceInstance, i)] = &cloudformation.AWSRDSDBInstance{
+			DBClusterIdentifier:  cloudformation.Ref(PostgresResourceCluster),
+			DBInstanceClass:      coalesce(p.Spec.AWS.InstanceType, DefaultClass),
+			Engine:               Engine,
+			PubliclyAccessible:   false,
+			DBParameterGroupName: cloudformation.Ref(PostgresResourceParameterGroup),
+			Tags:                 tags,
+			DBSubnetGroupName:    cloudformation.Ref(DBSubnetGroupNameParameterName),
+		}
+	}
+
+	template.Resources[PostgresResourceClusterParameterGroup] = &cloudformation.AWSRDSDBClusterParameterGroup{
+		Description: "GSP Service Operator Cluster Parameter Group",
+		Family:      Family,
+		Parameters: map[string]string{
+			"timezone": "UTC",
+		},
+		Tags: tags,
+	}
+
+	template.Resources[PostgresResourceParameterGroup] = &cloudformation.AWSRDSDBParameterGroup{
+		Description: "GSP Service Operator Parameter Group",
+		Family:      Family,
+		Parameters: map[string]string{
+			"application_name": p.GetStackName(),
+		},
+		Tags: tags,
+	}
+
+	template.Outputs[PostgresEndpoint] = map[string]interface{}{
+		"Description": "Postgres Endpoint used by the application to perform connection.",
+		"Value":       cloudformation.GetAtt(PostgresResourceCluster, "Endpoint.Address"),
+	}
+
+	template.Outputs[PostgresReadEndpoint] = map[string]interface{}{
+		"Description": "Postgres reader Endpoint used by the application to perform connection.",
+		"Value":       cloudformation.GetAtt(PostgresResourceCluster, "ReadEndpoint.Address"),
+	}
+
+	template.Outputs[PostgresPort] = map[string]interface{}{
+		"Description": "Postgres Port used by the application to perform connection.",
+		"Value":       cloudformation.GetAtt(PostgresResourceCluster, "Endpoint.Port"),
+	}
+
+	template.Outputs[PostgresUsername] = map[string]interface{}{
+		"Description": "Postgres master username",
+		"Value":       masterUsernameSecretRef,
+	}
+
+	template.Outputs[PostgresPassword] = map[string]interface{}{
```

Comment:
This is all sorts of scary. Can we trust that secrets manager will never try to interpret the resolve syntax in here?

Reply:
yes, it's a magic string thanks to secretsmanager